servlet/src/main/scala/org/http4s/servlet/Http4sServlet.scala

```diff
@@ -81,18 +81,28 @@ abstract class Http4sServlet[F[_]](
       servletResponse: HttpServletResponse,
       bodyWriter: BodyWriter[F],
   ): F[Unit] =
-    F.delay {
-      servletResponse.setStatus(response.status.code)
-      for (header <- response.headers.headers if header.name != ci"Transfer-Encoding")
-        servletResponse.addHeader(header.name.toString, header.value)
-    }.attempt
-      .flatMap {
+    // Note: the servlet API gives us no undeprecated method to both set
+    // a body and a status reason.  We sacrifice the status reason.
+    //
+    // This F.attempt.flatMap can be interrupted, which prevents the body from
+    // running, which prevents the response from finalizing.  Woe betide you if
+    // your effect isn't Concurrent.
+    F.uncancelable { poll =>
```

Comment:
I think you need `poll` around `bodyWriter(response)` and `response.body.drain....`. I also think you probably want to toss an `onCancel` on your `poll`s.

Reply:
Yes, `poll`ing `bodyWriter` and `response.body.drain` seems necessary.  But what would `onCancel` do?  `response.body.drain` perhaps, but what if `bodyWriter` got interrupted partway through, and `response.body`'s effects are not idempotent?

Again, this is all caused by an unfortunate design choice from before we had `Resource`.  But it's what we have in 0.23, and it affects every backend.

Reply:
That's also why we can't do something like:

```scala
Stream.bracket(acquireResponse)(_.body.drain.compile).flatMap(renderResponse)
```

We can make this as leakproof as 0.22 is and move on, but since we hit merge conflicts, it does seem like a nice opportunity to fix it.

Reply:
This is challenging because there isn't a strict separation of acquire-emit-release.  There are nested scopes.  We might "finalize a response" by calling `onFinalize` on the body, and then in a middleware wrap more effects.  The only way to release all resources acquired for the response but finalized in its body is to uncancelably drain the entire body.  Gross!

The right solution is still a resource of a response.  Something that could work in 0.23 would be a vault attribute that contains a finalizer.  Then a response would look much like a `Resource.Allocated` and be treated as such.  But it would be contrary to how we've told people to finalize resources since dinosaurs roamed the earth.

In any case, I don't think we can solve it the way I envisioned, nor should solve it in this merge.

Reply:
A lot to think about. But yeah, would be good to get this merge done, fix the website, and unblock the release train :)